packages/forklift-console-plugin/src/modules/Providers/modals/EditProviderURL/OpenstackEditURLModal.tsx

```diff
@@ -13,20 +14,65 @@ import { EditProviderURLModalProps } from './EditProviderURLModal';
 export const OpenstackEditURLModal: React.FC<EditProviderURLModalProps> = (props) => {
   const { t } = useForkliftTranslation();
 
-  const urlValidationHook: ValidationHookType = (value) => {
-    const isValidURL = validateURL(value.toString().trim());
+  const helperTextMsgs = {
+    error: (
+      <div className="forklift-edit-modal-field-error-validation">
+        <Trans t={t} ns="plugin__forklift-console-plugin">
+          Error: The format of the provided URL is invalid. Ensure the URL includes a scheme, a
+          domain name, and a path. For example:{' '}
+          <strong>https://identity_service.com:5000/v3</strong>.
+        </Trans>
+      </div>
+    ),
+    warning: (
+      <div className="forklift--edit-modal-field-warning-validation">
+        <Trans t={t} ns="plugin__forklift-console-plugin">
+          Warning: The provided URL does not end with the API endpoint path:
+          <strong>
+            {'"'}/v3{'"'}
+          </strong>
+          {'. '}
+          Ensure the URL includes the correct path. For example:{' '}
+          <strong>https://identity_service.com:5000/v3</strong>.
+        </Trans>
+      </div>
+    ),
+    success: (
+      <div className="forklift-edit-modal-field-success-validation">
+        <Trans t={t} ns="plugin__forklift-console-plugin">
+          For example: <strong>https://identity_service.com:5000/v3</strong>.
+        </Trans>
+      </div>
+    ),
+    default: (
+      <div className="forklift-edit-modal-field-default-validation">
+        <Trans t={t} ns="plugin__forklift-console-plugin">
+          {'For example: <strong>https://identity_service.com:5000/v3</strong>.'}
+        </Trans>
+      </div>
+    ),
+  };
 
-    return isValidURL
-      ? {
-          validationHelpText: undefined,
-          validated: 'success',
-        }
-      : {
-          validationHelpText: t(
-            'URL must start with https:// or http:// and contain valid hostname and path',
-          ),
-          validated: 'error',
-        };
+  const urlValidationHook: ValidationHookType = (value) => {
```

Comment:
urlValidationHook is a validator, it should be implemented in the validations part of our project, please move it to currect directory.

Reply:
This was not added as part of this PR, I just added more logic to that method. 
In addition, this validation method exists for all providers as part of the modal code (ocp, oVirt,vsphere), so better do that in a separate follow up PR

Reply:
"you touch it you own it", it was not in the right place before you added to it, and it's not in it's right place now. Since you added to it, can you move it to the currect place ?
p.s.
you will also need to rename to fit the validation methods naming convention

Reply:
I'm ok with a folow up PR to fix in all places this method is used out of place

Reply:
https://github.com/kubev2v/forklift-console-plugin/issues/864